src/Symfony/Bundle/FrameworkBundle/Resources/public/csrf_double_submit_controller.js

```diff
@@ -0,0 +1,34 @@
+document.addEventListener('submit', function (event) {
+    const csrfField = event.target.querySelector('input[data-controller="symfony--framework-bundle--csrf-double-submit"]');
+
+    if (!csrfField) {
+        return;
+    }
+
+    let csrfCookie = csrfField.getAttribute('data-symfony--framework-bundle--csrf-double-submit-cookie');
+
+    if (!csrfCookie) {
+        csrfField.setAttribute('data-symfony--framework-bundle--csrf-double-submit-cookie', csrfCookie = csrfField.value);
+    }
+
+
+    if (!/^[-a-zA-Z0-9_]+$/.test(csrfCookie)) {
+        return
+    }
+
+    let csrfToken = btoa(String.fromCharCode.apply(null, (window.crypto || window.msCrypto).getRandomValues(new Uint8Array(18))));
+    csrfField.value = csrfToken;
```

Comment:
```suggestion  
    // Alternative to btoa ?
    const bytesToHex = (bytes) => Array.from(bytes, byte => byte.toString(16).padStart(2, '0')).join('');
    const bytes = (window.crypto || window.msCrypto).getRandomValues(new Uint8Array(18));
    const csrfToken = bytesToHex(bytes);
    
    csrfField.value = csrfToken
```

As it is a base64 string, btoa should probably not be used here.



Also, [window.crypto](https://developer.mozilla.org/en-US/docs/Web/API/Window/crypto) is available for [99.3%](https://caniuse.com/?search=crypto) of users, and IE 11 is abandonned for years, so i'm not sure we need this here


```diff
- (window.crypto || window.msCrypto) 
+ window.crypto.
```

Reply:
We have the duty to provide a solution that works as broadly as possible. Considering the cost to support it, this should work on IE11 (it does right now)

Reply:
> As it is a base64 string, btoa should probably not be used here.

I don't understand this comment: it is b64-encoded *because* btoa is used.